### PR TITLE
Implement support for action workflows

### LINKED ISSUE #1399: Support GitHub Actions API
Announcement: https://developer.github.com/v3/actions/

### PR DESCRIPTION
**Pull request by joshuabezaleel**
This PR adds support for managing the most recent update of Action Workflows.
The branch was created from PR #1402 .

Relates to: #1399

### CODE REVIEW

File: github/actions_workflows.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return workflows, resp, nil
+}
+
+// GetWorkflow gets a specific workflow.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/workflows/#list-repository-workflows
```
Comment (gmlewis) [inline github/actions_workflows.go]: I think this URL should be:
https://developer.github.com/v3/actions/workflows/#get-a-workflow

File: github/actions_workflows.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	Path      string    `json:"path"`
+	State     string    `json:"state"`
+	CreatedAt Timestamp `json:"created_at"`
+	UpdatedAt Timestamp `json:"updated_at"`
+	URL       string    `json:"url"`
+	HTMLURL   string    `json:"html_url"`
+	BadgeURL  string    `json:"badge_url"`
+}
+
+// Workflows represents one item from the ListWorkflows response.
```
Comment (gmlewis) [inline github/actions_workflows.go]: How about:
```
// Workflows represents a slice of repository action workflows.
```

File: github/actions_workflows.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		return nil, resp, err
+	}
+
+	return workflows, resp, nil
+}
+
+// GetWorkflow gets a specific workflow.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/workflows/#list-repository-workflows
+func (s *ActionsService) GetWorkflow(ctx context.Context, owner, repo string, workflowID int64) (*Workflow, *Response, error) {
```
Comment (gmlewis) [inline github/actions_workflows.go]: How about call this one `GetWorkflowByID` and then make another one called `GetWorkflowByFileName` and in that second one change `workflowID int64` to `workflowFileName string`...
then both of these methods would call a common, private method where they passed in the `url` and then shared the rest of the body?

Something like this (sorry, it's probably easier to just write it out):

```
// GetWorkflowByID gets a specific workflow by ID.
//
// GitHub API docs: https://developer.github.com/v3/actions/workflows/#get-a-workflow
func (s *ActionsService) GetWorkflowByID(ctx context.Context, owner, repo string, workflowID int64) (*Workflow, *Response, error) {
  u := fmt.Sprintf("repos/%v/%v/actions/workflows/%v", owner, repo, workflowID)
  return s.getWorkflow(ctx, u)
}

// GetWorkflowByFileName gets a specific workflow by file name.
//
// GitHub API docs: https://developer.github.com/v3/actions/workflows/#get-a-workflow
func (s *ActionsService) GetWorkflowByFileName(ctx context.Context, owner, repo string, workflowFileName string) (*Workflow, *Response, error) {
  u := fmt.Sprintf("repos/%v/%v/actions/workflows/%v", owner, repo, workflowID)
  return s.getWorkflow(ctx, u)
}

func (s *ActionsService) getWorkflow(ctx context.Context, url string) (*Workflow, *Response, error) {
	req, err := s.client.NewRequest("GET", url, nil)
	if err != nil {
		return nil, nil, err
	}

	workflow := new(Workflow)
	resp, err := s.client.Do(ctx, req, workflow)
	if err != nil {
		return nil, resp, err
	}

	return workflow, resp, nil
}
```

**Comment by joshuabezaleel** [comment]
Thank you very much for the really great review, @gmlewis :slightly_smiling_face:

**Comment by gmlewis** [comment]
@martinssipenko  - if you are comfortable, please feel free to review this PR and then we can hopefully get it merged after you approve it and after we get #1402 merged.

**Comment by martinssipenko** [comment]
@googlebot I consent.

**Comment by martinssipenko** [comment]
LGTM

**Comment by gmlewis** [comment]
> LGTM

Thank you, @martinssipenko!  Awaiting the merge of #1402 now before merging this one.

**Comment by joshuabezaleel** [comment]
Thank you for the reviews, @martinssipenko @gauntface @gmlewis !! :tada: :tada: :tada: !!

**Comment by gmlewis** [comment]
Thanks, @gauntface.  I'm going to hold off merging this one until #1402 gets another LGTM and merged... as this is based off of that one.

File: github/actions_secrets.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+type Secrets struct {
+	TotalCount int       `json:"total_count"`
+	Secrets    []*Secret `json:"secrets"`
+}
+
+// ListSecrets lists all secrets available in a repository
+// without revealing their encrypted values.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/secrets/#list-secrets-for-a-repository
+func (s *ActionsService) ListSecrets(ctx context.Context, owner, repo string, opt *ListOptions) (*Secrets, *Response, error) {
```
Comment (gmlewis) [inline github/actions_secrets.go]: When you get a chance, could you please change `opt` to `opts` to be consistent with #1417?
(Here and anywhere else you find it in this PR, please.)

**Comment by gmlewis** [comment]
Thank you, @joshuabezaleel!
Could you please check one more time for the `opt` to `opts` now that we merged #1402?

**Comment by gmlewis** [comment]
(You will find them in files that you didn't previously modify, by the way.... and you might need to perform a "git merge master" into your branch to see them.)

**Comment by joshuabezaleel** [comment]
I hope that should do the trick. cc: @gmlewis

**Pull request merged**